core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -57,23 +58,98 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
-   * @param accum AccumProfileResults to be analyzed
+   * Updates the stageToDiagnosticMetrics mapping with the provided AccumProfileResults.
+   * @param accum AccumProfileResults instance containing diagnostic metrics to be added
+   *              to stageToDiagnosticMetrics mapping.
    */
   private def updateStageDiagnosticMetrics(accum: AccumProfileResults): Unit = {
     val stageId = accum.stageId
+
+    // Initialize an empty mapping for the stage if it doesn't already exist
     if (!stageToDiagnosticMetrics.contains(stageId)) {
       stageToDiagnosticMetrics(stageId) = HashMap.empty[String, AccumProfileResults]
     }
+
     stageToDiagnosticMetrics(stageId)(accum.accMetaRef.getName()) = accum
   }
 
+  /**
+   * Updates the IODiagnosticMetricsMap with the provided SQLAccumProfileResults.
+   * @param accum SQLAccumProfileResults instance containing IO diagnostics metrics
+   *              to be added to IODiagnosticMetricsMap.
+   */
+  private def updateIODiagnosticMetricsMap(accum: SQLAccumProfileResults): Unit = {
+    val key = (accum.sqlID, accum.nodeID, accum.stageIds)
+
+    // Initialize an entry if the key does not exist
+    if (!IODiagnosticMetricsMap.contains(key)) {
+      IODiagnosticMetricsMap(key) = ArrayBuffer[SQLAccumProfileResults]()
+    }
+
+    IODiagnosticMetricsMap(key) += accum
+  }
+
+  /**
+   * Retrieves the set of task IDs associated with a specific stage.
+   *
+   * @param stageId  The ID of the stage.
+   * @return         A set of task IDs corresponding to the given stage ID.
+   */
+  private def getStageTaskIds(stageId: Int): Set[Long] = {
+    app.taskManager.getAllTasksStageAttempt(stageId).map(_.taskId).toSet
+  }
+
+  /**
+   * Computes statistical metrics (min, median, max, sum) for task updates
+   * in a given stage based on a provided set of task IDs.
+   * @param accumInfo     AccumInfo object containing task update data.
+   * @param stageTaskIds  Set of task IDs corresponding to the stage.
+   * @return              Option containing a tuple of (min, median, max, sum)
+   *                      if there are task updates, or None if no updates exist.
+   */
+  private def getAccumInfoStatisticsInStage(accumInfo: AccumInfo, stageTaskIds: Set[Long]):
+      Option[(Long, Long, Long, Long)] = {
+    // Filter task updates to only include those matching the stage's task IDs
+    val filteredTaskUpdates =
+      accumInfo.taskUpdatesMap.filterKeys(stageTaskIds.contains).values.toSeq.sorted
+
+    if (filteredTaskUpdates.isEmpty) {
+      None
+    } else {
+      val min = filteredTaskUpdates.head
+      val max = filteredTaskUpdates.last
+      val sum = filteredTaskUpdates.sum
+      val median = if (filteredTaskUpdates.size % 2 == 0) {
+        val mid = filteredTaskUpdates.size / 2
+        (filteredTaskUpdates(mid) + filteredTaskUpdates(mid - 1)) / 2
+      } else {
+        filteredTaskUpdates(filteredTaskUpdates.size / 2)
+      }
+
+      Some((min, median, max, sum))
```

Comment:
By moving this code block into a separate method, the flattening of (min, max, med, sum) becomes useless.
When a tuple `(Long, Long, Long, Long)` is returned, this implies that scala is allocating an object of type tuple. In other words, the optimization done in previous Diagnostic-view-1 PR is rolled back by that change.

Reply:
thanks @amahussein! I will remove the function `getAccumInfoStatisticsInStage`